src/Datadog.Trace.ClrProfiler.Native/clr_helpers.cpp

```diff
@@ -233,25 +270,62 @@ std::vector<Integration> FilterIntegrationsByCaller(
   return enabled;
 }
 
+bool AssemblyMeetsIntegrationRequirements(
+    const AssemblyMetadata metadata,
+    const MethodReplacement method_replacement) {
+
+  const auto target = method_replacement.target_method;
+
+  if (target.assembly.name != metadata.name) {
+    // not the expected assembly
+    return false;
+  }
+
+  if (target.min_v_major > metadata.majorVersion) {
```

Comment:
I was thinking about how we could move this code that compares versions somewhere else and realized there's already a [`Version` struct](/DataDog/dd-trace-dotnet/blob/d21e37bd482cb15ea6bd5e97afa4f75e7f80b279/src/Datadog.Trace.ClrProfiler.Native/integration.h#L48) that we can probably use here. We can add relational operators to it (it already has an equality operator) and do comparisons like:
```cpp
// compares major, minor, etc
if (target.min_version <= metadata.version &&
    metadata.version <= target.max_version) {
   // version is inside valid range
}
```

We could also add a method to parse from a string, and maybe even a conversion from `ASSEMBLYMETADATA`.

Reply:
Yeah, that's perfect. Is it okay if I visit that in the next PR? I'll be applying this to all existing integrations per the docs, as well as adding exclusion reasons as logs.

Reply:
Sure. Thanks!